### ISSUE TITLE
As a project manager I would like to have a refactoring of the web application parking.bz.it and implement it on the base of existing / new web components that we can then publish on the web components store

### ISSUE THREAD

**Issue by rcavaliere** [noi-techpark/bz-parking-angular#17]
- [x] market search by @rcavaliere -> external supplier (Flashbeing)
- [x] Design of the UI: basically the same, but small improvements could be welcome
- [x] Implementation of the web component modules: the web-component parking could be used for the map visualization
- [x] Implementation of the web application
- [x] Publish the web application on the web components store

Demo available on https://parking.opendatahub.testingmachine.eu

**Comment by rcavaliere** [comment]
@dulvui can you please create a repository on GitHub so to host the source code of the new parking application?

**Comment by dulvui** [comment]
@rcavaliere Yes but how should the repo be named? Because there is already a webcomponent named webcomp-parking https://github.com/noi-techpark/webcomp-parking
Would webcomp-parking-forecast work, since this webcomponent also has the forecasts our what name would you prefer?

**Comment by rcavaliere** [comment]
@dulvui I would not rename it webcomp, but something link "parking-app" (it is developed using web components technologies, but is actually an application). We can then think to have another repo for the webcomponents components to be used for the store

**Comment by dulvui** [comment]
@rcavaliere Okay thanks now I understand. I created now https://github.com/noi-techpark/parking-app

**Comment by rcavaliere** [comment]
@dulvui PR has taken place: https://github.com/noi-techpark/parking-app/pull/1

**Comment by rcavaliere** [comment]
@dulvui can we deploy the new parking app version on an internal domain, e.g. http://parking.opendatahub.testingmachine.eu/, so that we can test it?

**Comment by dulvui** [comment]
@rcavaliere The app is now live on https://parking.opendatahub.testingmachine.eu/ for testing

**Comment by ohnewein** [comment]
Looks cool. There are some functions missing or not working, but I assume it's because this is a first alpha release. Correct?

Btw. don't forget to support the parameter to set the initial city. This is important for apps by cities like Merano to be able to include a link to it.

**Comment by rcavaliere** [comment]
> Looks cool. There are some functions missing or not working, but I assume it's because this is a first alpha release. Correct?
> 
> Btw. don't forget to support the parameter to set the initial city. This is important for apps by cities like Merano to be able to include a link to it.

Yes, it's still an alpha version. Thanks for the input, of course we will have this functionality implemented!

**Comment by rcavaliere** [comment]
@dulvui the new version of the parking APP is nice, but still too slow. Can you check if we can make some improvements in the way the API calls to the Open Data Hub are carried out?

**Comment by dulvui** [comment]
@rcavaliere I optimized now the mobility API calls but its still slow. The other parking webcomponent we have uses the exact same calls for mobility and tourism API and the stations show up instantly:
https://webcomponents.opendatahub.com/webcomponent/543ed85d-8280-4240-8625-4ee5102710ff

So in my opinion it is more a visualization problem, because on the other webcomponent it shows instantly.
Maybe adding a marker clustering or using another way to add the markers to the map would optimize the speed.

Should I investigate more to optimize the visualization?

**Comment by rcavaliere** [comment]
@dulvui thanks for the immediate feedback. Yes investigate a little, but we should just provide the feedback to Flashbeing on how they should fix this

**Comment by dulvui** [comment]
@rcavaliere It seems that there's a performance problem when using hundreds of markers with the library `vuelayers`.
Here there are closed issue about this problem:
https://github.com/ghettovoice/vuelayers/issues/355
And here more performance related closed issues:
https://github.com/ghettovoice/vuelayers/issues?q=is%3Aissue+is%3Aclosed+performance

It seams that using `vl-feature` is not recommended when a lot of markers should be visible.
They recommend using a features array. But My knowledge of nuxtjs and this library is to little to fix this problem.
But with this information, Flasbeing should be able to fix this performance problems.

**Comment by rcavaliere** [comment]
@dulvui thanks, I will forward these inputs to Flashbeing!

**Comment by rcavaliere** [comment]
@dulvui @ohnewein Flashbeing has tried to improve the performance, we received a new PR. In my opinion, it is better but still not sufficient, but hearing what Simon said the issue is that vuelayers, despite our indications, continues to be used. I don't know if it is more efficient if we invest a couple of hours of our time trying to implement solution for that, for example integrating the same mechanism of markers visualization on the map used on the existing web component

**Comment by dulvui** [comment]
@rcavaliere Yes, I agree that the performance didn't improve enough.
The difference now is, that only visible markers are shown, so if the zoom level is set to a city, only a few markers need to be shown and it loads fast. But if you zoom out to whole South Tyrol the performance issue remains as before.
I recommended removing `vl-feature` and following the instructions o improve the performance, but it is still used in the new PR. Removing `vl-feature` should solve the problem and a switch to another library is not needed.
I can invest some hours this week and try to fix this if you want. Let me know.

**Comment by rcavaliere** [comment]
@dulvui I don't know how much you are loaded in these days, if you manage to this and the effort can be kept to a reasonable amount of hours I would do it

**Comment by dulvui** [comment]
@rcavaliere Now I was able to remove `vl-feature` and use `vl-source-vector` and the performance problem is solved with this. Now only the styling is missing. I will let you know, when I fixed also the styling of the markers.

**Comment by rcavaliere** [comment]
@dulvui can you please let your improvement of the application available on a different end-point with respect to current application version developed by Flashbeing? We would like to have them available in order to compare them

**Comment by dulvui** [comment]
@rcavaliere Yes, now 2 versions are online:
- https://parking.opendatahub.testingmachine.eu/ by FlashBeing
- https://parking2.opendatahub.testingmachine.eu/ with my changes

**Comment by rcavaliere** [comment]
@dulvui super work! Am I wrong or in your application version you are using a demo marker icon? In case yes, can you the same icons as Flashbeing uses?

**Comment by dulvui** [comment]
Yes, I was only able to show the default marker, but the styling like FlashBeing is still missing.
I will try tomorrow again to adjust also the styling

**Comment by dulvui** [comment]
@rcavaliere I fixed now the styling and it is still not exactly the same as FlashBeing's one but as PoC that this approach has a better performance is now confirmed.
I'll try to make the style exactly the same in the next days.

**Comment by rcavaliere** [comment]
For the web-component part @dulvui will prepare what necessary for Flashbeing

**Comment by rcavaliere** [comment]
@dulvui please integrate this new PR by Flashbeing: https://github.com/noi-techpark/parking-app/pull/6
Please also check why we don't provide through the API the field smetadata.standard_name. 
Once the APP is consolidated, let's then proceed with the web-component preparation

**Comment by dulvui** [comment]
@rcavaliere I merged now the PR and will check the standard_name issue

**Comment by dulvui** [comment]
@rcavaliere standard_name is now working again

**Comment by rcavaliere** [comment]
@dulvui that's good, standard_name is working. The application however does not seem to have improved that much, on Chrome the feeling is little better than on Firefox. What do you think @dulvui and @ohnewein ?

**Comment by dulvui** [comment]
@rcavaliere I was not able to deploy the latest version yet, because the build is failing. I'll let you know when its online in testing.

**Comment by rcavaliere** [comment]
@dulvui ah ok.... that's the explanation :-)

**Comment by rcavaliere** [comment]
@ohnewein @dulvui I think that with the new contribution we can now close this development and put everything in production. Remember that we have the application, to be reachable through the domain parking.bz.it, and the web-component, to be integrated in the productive web-component store. Of course after Eastern!
